docs/how-to/topics/file-management.rst

```diff
@@ -0,0 +1,206 @@
+- On Unix, even if ``sudo`` is used to install the app, it installs to a directory that the user does not have permissions to write to.
+
+You can read from ``paths.app``, but you shouldn't write to it.
+
+So, what if you want to generate a file through your app and save it? Toga provides four writable paths available for storing files associated with an app:
+
+- ``data``: The location for storing user data.
+- ``config``: The location for storing user configuration data.
+- ``cache``: The location for storing cache files. This should be used only for easily regenerated files as the operating system may purge the contents of this directory without warning.
+- ``logs``: The location for storing log files.
+
+These paths are different on every operating system, and Toga guarantees the correct paths will be provided. The paths will be subdirectories found in ``~/Library`` on macOS, XDG-compliant dotfiles in ``~`` on Linux, and the user's ``AppData`` directory on Windows.
+
+Let's build on the current application to generate a configuration file from the contents of the ``initial_config.toml`` file.
+
+Update the ``save_button_pressed`` handler in ``app.py`` to the following:
+
+.. code-block:: python
+
+     def save_button_pressed(self, button, **kwargs):
+        path = self.paths.config / "config.toml"
+        self.config_directory.value = path
+        path.write_text(self.text_input.value, encoding="utf-8")
+
+This change implements the save button, that when pressed, saves the contents of the text input to a ``config.toml`` file in an app-specific subdirectory of the operating-system appropriate configuration directory, and displays the path to the file below the input.
+
+Run the app and click the "Load initial configuration" button to load the file contents into the text input. Update the variables to whatever you like. Click the save button to generate the file. In your file explorer or terminal, you can use the path displayed below the input to find and view your new configuration file directly.
+
+Updating an existing file
+=========================
+
+Now that the configuration file is generated, you may want to update it. You could use the same app to load the contents of ``initial_config.toml`` and update that info to the new configuration, but then you may not know what the previous changes were. Instead, you can tell the app to check for an existing configuration file, and load the contents of that if it exists.
+
+Update the ``load_button_pressed`` handler in ``app.py`` to the following:
+
+.. code-block:: python
+
+    def load_button_pressed(self, button, **kwargs):
+        path = self.paths.config / "config.toml"
+        if not path.exists():
+            path = self.paths.app / "resources/initial_config.toml"
```

Comment:
Discrepancy here in `"resources/initial_config.toml"` vs `"resources" / "initial_config.toml"`. Both are valid, but we should be consistent.

Reply:
This was from the previous version. I had it updated in the actual Python file for this, but missed updating it here. Fixing.